cogs/annual_handover_and_reset.py

```diff
@@ -222,3 +227,109 @@ async def annual_roles_reset(self, ctx: TeXBotApplicationContext) -> None:
+            content=":hourglass: Processing year channel iteration... :hourglass:",
+        )
+
+        final_year_channel: discord.TextChannel | None = discord.utils.get(
+            main_guild.text_channels,
+            name="final-years",
+        )
+
+        if final_year_channel:
+            await initial_message.edit(
+                content=":hourglass: Archiving final year channel... :hourglass:",
+            )
+            archivist_role: discord.Role = await self.bot.archivist_role
+
+            await final_year_channel.set_permissions(guest_role, overwrite=None)
+            await final_year_channel.set_permissions(archivist_role, read_messages=True)
+
+            await final_year_channel.edit(
+                name="final-years-" + str(datetime.datetime.now(tz=datetime.UTC).year),
+            )
+
+            archived_category: discord.CategoryChannel | None = discord.utils.get(
+                main_guild.categories,
+                name="Archived",
+            )
+
+            if archived_category:
+                await final_year_channel.edit(
+                    category=archived_category,
+                    sync_permissions=True,
+                )
+
+        second_year_channel: discord.TextChannel | None = discord.utils.get(
+            main_guild.text_channels,
+            name="second-years",
+        )
+
+        if second_year_channel:
+            await second_year_channel.edit(name="final-years")
+
+        first_year_channel: discord.TextChannel | None = discord.utils.get(
+            main_guild.text_channels,
+            name="first-years",
+        )
+
+        if first_year_channel:
+            await first_year_channel.edit(name="second-years")
+
+        year_channels_category: discord.CategoryChannel | None = discord.utils.get(
+            main_guild.categories,
+            name="Year Chats",
+        )
+
+        await initial_message.edit(content=(
+            ":hourglass: Creating new first year channel and setting permissions... "
+            ":hourglass:"
+        ))
+
+        new_first_year_channel: discord.TextChannel = await main_guild.create_text_channel(
+            name="first-years",
+        )
+
+        if year_channels_category:
+            await new_first_year_channel.edit(
+                category=year_channels_category,
+                sync_permissions=True,
+                position=0,
+            )
+
+            await initial_message.edit(
+                content=":white_check_mark: Year channel iterations complete!",
+            )
+            return
+
+        await new_first_year_channel.set_permissions(
+            guest_role,
+            read_messages=True,
+            send_messages=True,
+        )
+
+        await initial_message.edit(content=(
+            ":white_check_mark: Year channel iterations complete "
+            "but no year channel category was found!"
+        ))
```

Comment:
Only one success message responder should be present, to improve legibility.

Fix:
```python
if year_channels_category:
    await new_first_years_channel.edit(
        category=year_channels_category,
        sync_permissions=True,
        position=0,
    )

else:
    await new_first_years_channel.set_permissions(
        guest_role,
        read_messages=True,
        send_messages=True,
    )

await initial_message.edit(
    content=(
        ":white_check_mark: Year channels were successfully incremented"
        f"{
            ". However, no \"Year Chats\" category could be found"
            if not year_channels_category else
            ""
        }. :white_check_mark:"
    ),
)
```